types/webpack/index.d.ts

```diff
@@ -646,6 +647,118 @@ declare namespace webpack {
             /** Generate records with relative paths to be able to move the context folder". */
             portableRecords?: boolean;
         }
+        interface DevServerOptions {
+            /** Provides the ability to execute custom middleware after all other middleware internally within the server. */
+            after?: () => void;
+            /** This option allows you to whitelist services that are allowed to access the dev server. */
+            allowedHosts?: string[];
+            /** Provides the ability to execute custom middleware prior to all other middleware internally within the server. */
+            before?: () => void;
+            /** This option broadcasts the server via ZeroConf networking on start. */
+            bonjour?: boolean;
+            /**
+             * When using inline mode, the console in your DevTools will show you messages e.g. before reloading,
+             * before an error or when Hot Module Replacement is enabled. This may be too verbose.
+             */
+            clientLogLevel?: 'none' | 'error' | 'warning' | 'info';
+            /** Enable gzip compression for everything served. */
+            compress?: boolean;
+            /**
+             * Tell the server where to serve content from. This is only necessary if you want to serve static files.
+             * devServer.publicPath will be used to determine where the bundles should be served from, and takes precedence.
+             */
+            contentBase?: boolean | string | string[];
+            /**
+             * When set to true this option bypasses host checking.
+             * THIS IS NOT RECOMMENDED as apps that do not check the host are vulnerable to DNS rebinding attacks.
+             */
+            disableHostCheck?: boolean;
+            /**
+             * This option lets you reduce the compilations in lazy mode.
+             * By default in lazy mode, every request results in a new compilation.
+             * With filename, it's possible to only compile when a certain file is requested.
+             */
+            filename?: string;
+            /** Adds headers to all responses. */
+            headers?: object;
+            /** When using the HTML5 History API, the index.html page will likely have to be served in place of any 404 responses. */
+            historyApiFallback?: boolean | object;
+            /** Specify a host to use. By default this is localhost. */
+            host?: string;
+            /** Enable webpack's Hot Module Replacement feature. */
+            hot?: boolean;
+            /** Enables Hot Module Replacement (see devServer.hot) without page refresh as fallback in case of build failures. */
+            hotOnly?: boolean;
+            /** By default dev-server will be served over HTTP. It can optionally be served over HTTP/2 with HTTPS. */
+            https?: boolean | object;
```

Comment:
Maybe instead of object create an interface?

https://webpack.js.org/configuration/dev-server/#devserver-https